src/utils/dashboardModelUtils.ts

```diff
@@ -0,0 +1,142 @@
+import { RootNodeModel, SectionNodeModel, WidgetModel } from '../interfaces/nodeModels';
+
+
+export function removeWidgetFromState(contents: any, id: string) {
+  // Helper to remove a widget from contents that get passed in
+
+  // This is composed of 2 Steps:
+
+  /*
+      Step 1:
+        - Find the section that the widget is contained in
+        - Remove it from the section's children
+        - Adjust the relative sizing of the section's remaining children
+        - If the sections children are 0 or 1, we need to do some adjustments
+        - If the above is the case call removeSection to handle this
+
+      Step 2:
+        - Remove the Widget from the Widgets list
+  */
+
+  for(let section in contents.sections) {
+    if(contents.sections[section].children.includes(id)) {
+
+      // Call helper to remove the child from the section and redistribute
+      // sizing amongst the remaining children
+      contents = removeChildAndReDistributeSizing(contents, section, id);
+
+      // IF remaining children is 0 or 1 we have to do some extra stuff
+      // This will be handled by the 'removeSection' function
+      if(contents.sections[section].children.length <= 1) {
+        contents = removeSection(contents, contents.widgets[id].parentId);
```

Comment:
as a result of my immutable comment this should probably only modify the `sections` object